atlasdb-config/src/main/java/com/palantir/atlasdb/factory/PersistentStorageFactories.java

```diff
@@ -68,23 +73,50 @@ public static synchronized void sanitizeStoragePath(String storagePath) {
                 "Storage path has to point to a directory",
                 SafeArg.of("storageDirectory", storageDirectory.getAbsolutePath()));
 
-        for (File file : MoreObjects.firstNonNull(storageDirectory.listFiles(), new File[0])) {
-            if (file.isDirectory() && (UUID_PATTERN.matcher(file.getName()).matches())) {
-                try {
-                    deletePath(file.toPath().toAbsolutePath());
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                }
+        List<File> uuidSubDirectories = uuidNamedSubDirectories(
+                MoreObjects.firstNonNull(storageDirectory.listFiles(), new File[0]));
+        List<File> directoryContentToDelete = limitFoldersToDelete(uuidSubDirectories);
+        deleteDirectories(directoryContentToDelete);
+
+        SANITIZED_PATHS.add(storagePath);
+    }
+
+    private static List<File> uuidNamedSubDirectories(File[] files) {
+        return Arrays.stream(files)
+                .filter(PersistentStorageFactories::folderFilter)
+                .collect(Collectors.toList());
+    }
+
+    private static boolean folderFilter(File file) {
+        return file.isDirectory() && (UUID_PATTERN.matcher(file.getName()).matches());
+    }
+
+    private static List<File> limitFoldersToDelete(List<File> folders) {
+        if (folders.size() > DEFAULT_DELETION_LIMIT) {
+            log.warn(
+                    "You are trying to delete more that two UUID named folders during persistent storage start-up. "
+                            + "Only two will be deleted.",
+                    SafeArg.of("number", folders.size()));
+        }
+        return folders.stream().limit(DEFAULT_DELETION_LIMIT).collect(Collectors.toList());
```

Comment:
I think I'd prefer to delete 0 if there are >2 since the most likely state of the world then is that you're in a wrong directory

Reply:
Now deletes 0.